src/librbd/cache/pwl/Types.h

```diff
@@ -278,32 +283,30 @@ struct WriteLogCacheEntry {
 };
 
 struct WriteLogPoolRoot {
-  #ifdef WITH_RBD_RWL
-  union {
-    struct {
-      uint8_t layout_version;    /* Version of this structure (RWL_POOL_VERSION) */
-    };
-    uint64_t _u64;
-  } header;
-  TOID(struct WriteLogCacheEntry) log_entries;   /* contiguous array of log entries */
-  #endif
-  #ifdef WITH_RBD_SSD_CACHE
-  uint64_t layout_version = 0;
-  uint64_t cur_sync_gen = 0;
-  #endif
+  uint64_t layout_version;
```

Comment:
this looks fine to me. AFAIK the `_u64` variable wasn't used in `rwl` mode

Reply:
@CongMinYin would be good to update here as well ` uint8_t layout_version;`

Reply:
Setting it to `uint8_t`  here does not save space. The previous code also used `uint64_t`. In a structure, there is a byte alignment rule. Because the next element is `uint64_t`, it will be aligned to an integer multiple address of 64 bits. Therefore, even if the version is `uint8_t`, it will occupy 64 bits of space. Because next  element will start from 64 bit aligned address. so why don't use `uint64_t`.

Reply:
I see, should be okay then, thanks for the explaination!